### PR TITLE
Extract IO-related functionalities as io.Manager

### PR DESCRIPTION
**Pull request by kzys**
The shim's service struct is getting bigger and bigger. This change
extracts IO-related functionalities as io.Manager to make the service
struct slightly smaller.

Signed-off-by: Kazuyoshi Kato <katokazu@amazon.com>

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: runtime/io/manager.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+const (
+	// taskExecID is a special exec ID that is pointing its task itself.
+	// While the constant is defined here, the convention is coming from containerd.
+	TaskExecID = ""
+
+	minVsockIOPort             = uint32(11000)
+	defaultVSockConnectTimeout = 5 * time.Second
+)
+
+type Manager struct {
```
Comment (kzys) [inline runtime/io/manager.go:38]: I don't like the name, but not having any better alternatives...

File: runtime/io/manager.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	}
+	return nil
+}
+
+// Find returns FIFO paths regarding the given exec.
+func (m *Manager) Find(taskID, execID string) cio.Config {
+	m.fifosMu.Lock()
+	defer m.fifosMu.Unlock()
+
+	return m.fifos[taskID][execID]
```
Comment (IRCody) [inline runtime/io/manager.go:126]: Is it not possible here for taskId or execId to not be found?

Comment (kzys) [reply runtime/io/manager.go:126]: Good point. I think it wouldn't be but I'd like to have the check just in case.

**Comment by kzys** [comment]
#483 changed a lot of stuff around IO handling. I'd like to step back and think about the issue more.

**Pull request closed without merging**